encodings/runend/src/compute.rs

```diff
@@ -160,6 +160,8 @@ impl FilterFn for RunEndArray {
 // Code adapted from apache arrow-rs https://github.com/apache/arrow-rs/blob/b1f5c250ebb6c1252b4e7c51d15b8e77f4c361fa/arrow-select/src/filter.rs#L425
 fn filter_run_ends<R: NativePType + AddAssign + From<bool> + AsPrimitive<u64>>(
     run_ends: &[R],
```

Comment:
I tried converting this to `impl Iterator<Item = usize>` but still need to pass the length and would needlesly widen type of the ends.

Reply:
You can do impl Iterator<usize> + ExactSizeIterator